Dual Battery Support.md

```diff
@@ -0,0 +1,262 @@
+                       If (\BT2T)
+                       {
+                           Local0 = \_SB.PCI0.LPC.EC.BAT1.XB1S
+                           If ((HB1A && !SLUL))
+                           {
+                               Local1 = 0x01
+                           }
+                           Else
+                           {
+                               Local1 = 0x00
+                           }
+
+                           If ((Local0 ^ Local1))
+                           {
+                               \_SB.PCI0.LPC.EC.BAT1.XB1S = Local1
+                               Notify (\_SB.PCI0.LPC.EC.BAT1, 0x01) // Device Check
+                           }
+                       }
+                   }
+```
+
+As you can see here, the Methods that we need to patch are:
+
+```
+Method (_Q22)
+Method (_Q4A)
+Method (_Q4B)
+Method (_Q4D)
+Method (_Q24)
+Method (_Q25)
+Method (_BATW)
+```
+
+So first step we need to do is to rename the native Method and place the patched one into a separate SSDT, example:
+`Method (_Q22)` can be renamed to `XQ22` from **config.plist > ACPI > Patch >**
+
+**Find: 5F513232 00**
+(5F513232 00 is _Q22 translated into hex code)
+
+**Replace: 58513232 00**
+(58513232 00 is XQ22 translated into hex code)
+
+Do the renames for all the found methods that will need to be patched like in the example below:
+(keep in mind to check if code with similar name doesn't exist in ACPI/DSDT, if yes, use different name)
+
+```
+Method (_Q22) to Method XQ22
+Method (_Q4A) to Method XQ4A
+Method (_Q4B) to Method XQ4B
+Method (_Q4D) to Method XQ4D
+Method (_Q24) to Method XQ24
+Method (_Q25) to Method XQ25
+Method (_BATW) to Method XATW
+```
+
+(keep in mind to check if code with similar name doesn't exist in ACPI/DSDT, if yes, use different name)
+
+Next we copy the original `Method (_Q22)` from DSDT and paste it into a new SSDT:
+
+```
+Method (_Q22, 0, NotSerialized)  // _Qxx: EC Query, xx=0x00-0xFF
+{
+    CLPM ()
+    If (HB0A)
+    {
+        Notify (BAT0, 0x80) // Status Change
+    }
+    If (HB1A)
+    {
+        Notify (BAT1, 0x80) // Status Change
+    }
+}
+```
+
+Now we rename Notifiers from `BAT0` and `BAT1` to `BATC`:
+
+```
+Method (_Q22, 0, NotSerialized)  // _Qxx: EC Query, xx=0x00-0xFF
+{
+    CLPM ()
+    If (HB0A)
+    {
+        Notify (BATC, 0x80) // Status Change
+    }
+    If (HB1A)
+    {
+        Notify (BATC, 0x80) // Status Change
+    }
+}
+```
+
+Next we add the `If (_OSI ("Darwin"))`.
+This makes sure that the patched code does get executed only if the operating system is macOS ("Darwin").
+So now the code will execute on macOS but won't in other Operating Systems like Windows or Linux, etc.
+For this we add the Else into the code and call the original method that we renamed to Method XQ22.
+By this, if macOS, the patched SSDT will load, if it's something else like Windows or Linux it will load the original method that we renamed to XQ22 in our case, like shown below:
```

Comment:
All SSDTs will *load* in all operating systems. The original method will be *executed* on Windows and Linux, as the `If (_OSI ("Darwin"))` is a runtime condition.

Reply:
Ok will cleanup this section as well

Reply:
@usr-sse2 edited this section on latest commit, let me know if it needs more tweaking.